doc/source/whatsnew/v3.0.0.rst

```diff
@@ -239,6 +239,7 @@ Other enhancements
 - Added a new :meth:`DataFrame.from_arrow` method to import any Arrow-compatible
   tabular data object into a pandas :class:`DataFrame` through the
   `Arrow PyCapsule Protocol <https://arrow.apache.org/docs/format/CDataInterface/PyCapsuleInterface.html>`__ (:issue:`59631`)
+- :meth:`DataFrame.itertuples` is now annotated to return an iterable of Any to accommodate dynamic namedtuple types
```

Comment:
pandas publishes type-hints through `pandas-stubs`, the ones in the pandas library itself do not impact users. As this is not a user-facing change, the whatsnew note should be removed.